lib/krane/deprecated_deploy_task.rb

```diff
@@ -447,6 +445,32 @@ def deploy_resources(resources, prune: false, verify:, record_summary: true)
       end
     end
 
+    def replace_or_create_resource(resource, force: false)
+      args = if force
+        ["replace", "--force", "--cascade", "-f", resource.file_path]
+      else
+        ["replace", "-f", resource.file_path]
+      end
+
+      _, _, status = kubectl.run(*args, log_failure: false)
+      return [nil, status] if status.success?
+
+      # it doesn't exist so we can't replace it, we try to create it
+      create_resource(resource)
+    end
+
+    def create_resource(resource)
+      out, err, status = kubectl.run("create", "-f", resource.file_path, log_failure: false, output: 'json')
+
+      # For resources that have a deploy_method of :create, we get the `name` from the result of the call to `create`
+      # We must explicitly set this name value so that the `apply` step for pruning can run successfully
+      if status.success? && resource.deploy_method == :create
+        resource.use_generate_name(out)
```

Comment:
nit `use_generated_name` (generateName is the name of the field, and we want the name the server already generated from it, not the value of that field)